### ISSUE TITLE
Not sure that this is an issue - unable to hook it up to jack

### ISSUE THREAD

**Issue by siborg** [Drumfix/avb4linux#11]
Hello, not sure I am doing something wrong but i get the avb connection up, it is picked up on the motu interface and also visible as a pcm device in alsa. I cannot seem to be able to configure Jack to utilise this pcm device. Are there are any special considerations when configuring it with Jack?

**Comment by ericfont** [comment]
what is the console output from when you try to stark jack?  (in Cadence, under Tools tab, press "Logs".  Or in QJackCtl press "Messages".)

**Comment by siborg** [comment]
> Tue Apr  6 22:48:44 2021: creating alsa driver ... hw:AVB,0|hw:AVB,0|256|3|96000|8|8|hwmon|hwmeter|-|32bit
> Tue Apr  6 22:48:44 2021: configuring for 96000Hz, period = 256 frames (2.7 ms), buffer = 3 periods
> Tue Apr  6 22:48:44 2021: ALSA: final selected sample format for capture: 32bit integer little-endian
> Tue Apr  6 22:48:44 2021: ALSA: use 3 periods for capture
> Tue Apr  6 22:48:44 2021: ALSA: final selected sample format for playback: 32bit integer little-endian
> Tue Apr  6 22:48:44 2021: ALSA: use 3 periods for playback
> Tue Apr  6 22:48:48 2021: ERROR: ALSA: poll time out, polled for 3999130 usecs, Retrying with a recovery, retry cnt = 1
> Tue Apr  6 22:48:49 2021: ERROR: JackPosixProcessSync::LockedTimedWait error usec = 5000000 err = Connection timed out
> Tue Apr  6 22:48:49 2021: ERROR: Driver is not running
> Tue Apr  6 22:48:49 2021: ERROR: Cannot open client name = dbusapi
> Tue Apr  6 22:48:49 2021: ERROR: failed to create dbusapi jack client
> Tue Apr  6 22:48:52 2021: ERROR: ALSA: poll time out, polled for 3999575 usecs, Retrying with a recovery, retry cnt = 2
> Tue Apr  6 22:48:56 2021: ERROR: ALSA: poll time out, polled for 3999492 usecs, Retrying with a recovery, retry cnt = 3
> Tue Apr  6 22:49:00 2021: ERROR: ALSA: poll time out, polled for 3999419 usecs, Retrying with a recovery, retry cnt = 4
> Tue Apr  6 22:49:04 2021: ERROR: ALSA: poll time out, polled for 3999358 usecs, Retrying with a recovery, retry cnt = 5
> Tue Apr  6 22:49:08 2021: ERROR: ALSA: poll time out, polled for 3999302 usecs, Reached max retry cnt = 5, Exiting
> Tue Apr  6 22:49:08 2021: ERROR: JackAudioDriver::ProcessAsync: read error, stopping...
> Tue Apr  6 22:49:08 2021: Released audio card Audio7
> Tue Apr  6 22:49:08 2021: Saving settings to "/home/skosecki/.config/jack/conf.xml" ...

The same jack client works ok with USB devices.

**Comment by ericfont** [comment]
I'm curious have you been able to use the PCM AVB ALSA device as a regular ALSA device without starting Jack?  For instance, try starting Audacity, and have Audacity in ALSA mode, and then for input and output device select the ALSA AVB device and set number capture channels up to 8 and press record, and then try to playback through the ALSA AVB.

**Comment by ericfont** [comment]
For the record, I'm also able to start in 96000 kHz with 256-frames/period and 3 periods/buffer:

```
Tue Apr  6 18:04:49 2021: Starting jack server...
Tue Apr  6 18:04:49 2021: JACK server starting in realtime mode with priority 7
Tue Apr  6 18:04:49 2021: self-connect-mode is "Don't restrict self connect requests"
Tue Apr  6 18:04:49 2021: Acquired audio card Audio4
Tue Apr  6 18:04:49 2021: creating alsa driver ... hw:AVB,0|hw:AVB,0|256|3|96000|8|8|nomon|swmeter|-|32bit
Tue Apr  6 18:04:49 2021: configuring for 96000Hz, period = 256 frames (2.7 ms), buffer = 3 periods
Tue Apr  6 18:04:49 2021: ALSA: final selected sample format for capture: 32bit integer little-endian
Tue Apr  6 18:04:49 2021: ALSA: use 3 periods for capture
Tue Apr  6 18:04:49 2021: ALSA: final selected sample format for playback: 32bit integer little-endian
Tue Apr  6 18:04:49 2021: ALSA: use 3 periods for playback
Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,0 in-hw-0-0-0-Ozonic-MIDI-1
Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,1 in-hw-0-0-1-Ozonic-MIDI-2
Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,0 out-hw-0-0-0-Ozonic-MIDI-1
Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,1 out-hw-0-0-1-Ozonic-MIDI-2
Tue Apr  6 18:04:49 2021: scan: added port hw:3,0,0 in-hw-3-0-0-MPKmini2-MIDI-1
Tue Apr  6 18:04:49 2021: scan: added port hw:3,0,0 out-hw-3-0-0-MPKmini2-MIDI-1
18:04:51.130 D-BUS: JACK server was started (org.jackaudio.service aka jackdbus).
```

though mine has `|nomon|swmeter|` while yours has `|hwmon|hwmeter|`

**Comment by ericfont** [comment]
though I'm also able to use `|hwmon|hwmeter|` too:

```
Tue Apr  6 18:08:36 2021: Starting jack server...
Tue Apr  6 18:08:36 2021: JACK server starting in realtime mode with priority 7
Tue Apr  6 18:08:36 2021: self-connect-mode is "Don't restrict self connect requests"
Tue Apr  6 18:08:36 2021: Acquired audio card Audio4
Tue Apr  6 18:08:36 2021: creating alsa driver ... hw:AVB,0|hw:AVB,0|256|3|96000|8|8|hwmon|hwmeter|-|32bit
Tue Apr  6 18:08:36 2021: configuring for 96000Hz, period = 256 frames (2.7 ms), buffer = 3 periods
Tue Apr  6 18:08:36 2021: ALSA: final selected sample format for capture: 32bit integer little-endian
Tue Apr  6 18:08:36 2021: ALSA: use 3 periods for capture
Tue Apr  6 18:08:36 2021: ALSA: final selected sample format for playback: 32bit integer little-endian
Tue Apr  6 18:08:36 2021: ALSA: use 3 periods for playback
```

**Comment by ericfont** [comment]
One common cause of error is that some other program has control over the interface when you start jack.  If I had another program using the device and try to start jack, jack fails with:

```
Tue Apr  6 18:14:55 2021: Starting jack server...
Tue Apr  6 18:14:55 2021: JACK server starting in realtime mode with priority 7
Tue Apr  6 18:14:55 2021: self-connect-mode is "Don't restrict self connect requests"
Tue Apr  6 18:14:55 2021: Acquired audio card Audio4
Tue Apr  6 18:14:55 2021: creating alsa driver ... hw:AVB|hw:AVB|18|2|44100|8|8|hwmon|swmeter|-|32bit
Tue Apr  6 18:14:55 2021: ERROR: 
ATTENTION: The capture device "hw:AVB" is already in use. The following applications  are using your soundcard(s) so you should  check them and stop them as necessary before  trying to start JACK again:
jackdbus (process ID 2723)
audacity (process ID 2
Tue Apr  6 18:14:55 2021: Released audio card Audio4
Tue Apr  6 18:14:55 2021: ERROR: Cannot initialize driver
Tue Apr  6 18:14:55 2021: ERROR: JackServer::Open failed with -1
Tue Apr  6 18:14:55 2021: ERROR: Failed to open server
Tue Apr  6 18:14:57 2021: Saving settings to "/home/e/.config/jack/conf.xml" ...
18:14:58.774 Could not connect to JACK server as client. - Overall operation failed. - Unable to connect to server. Please check the messages window for more info.
```

So maybe double check you don't  have something like PulseAudio taking over the device...and if you do, the user pavucontrol and in "Hardware" tab, click the AVB device, and set its profile to "Off".

**Comment by siborg** [comment]
> For the record, I'm also able to start in 96000 kHz with 256-frames/period and 3 periods/buffer:
> 
> ```
> Tue Apr  6 18:04:49 2021: Starting jack server...
> Tue Apr  6 18:04:49 2021: JACK server starting in realtime mode with priority 7
> Tue Apr  6 18:04:49 2021: self-connect-mode is "Don't restrict self connect requests"
> Tue Apr  6 18:04:49 2021: Acquired audio card Audio4
> Tue Apr  6 18:04:49 2021: creating alsa driver ... hw:AVB,0|hw:AVB,0|256|3|96000|8|8|nomon|swmeter|-|32bit
> Tue Apr  6 18:04:49 2021: configuring for 96000Hz, period = 256 frames (2.7 ms), buffer = 3 periods
> Tue Apr  6 18:04:49 2021: ALSA: final selected sample format for capture: 32bit integer little-endian
> Tue Apr  6 18:04:49 2021: ALSA: use 3 periods for capture
> Tue Apr  6 18:04:49 2021: ALSA: final selected sample format for playback: 32bit integer little-endian
> Tue Apr  6 18:04:49 2021: ALSA: use 3 periods for playback
> Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,0 in-hw-0-0-0-Ozonic-MIDI-1
> Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,1 in-hw-0-0-1-Ozonic-MIDI-2
> Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,0 out-hw-0-0-0-Ozonic-MIDI-1
> Tue Apr  6 18:04:49 2021: scan: added port hw:0,0,1 out-hw-0-0-1-Ozonic-MIDI-2
> Tue Apr  6 18:04:49 2021: scan: added port hw:3,0,0 in-hw-3-0-0-MPKmini2-MIDI-1
> Tue Apr  6 18:04:49 2021: scan: added port hw:3,0,0 out-hw-3-0-0-MPKmini2-MIDI-1
> 18:04:51.130 D-BUS: JACK server was started (org.jackaudio.service aka jackdbus).
> ```
> 
> though mine has `|nomon|swmeter|` while yours has `|hwmon|hwmeter|`

Hello, 

No it does not seem to play anything when I try via Audacity

**Comment by siborg** [comment]
> I'm curious have you been able to use the PCM AVB ALSA device as a regular ALSA device without starting Jack? For instance, try starting Audacity, and have Audacity in ALSA mode, and then for input and output device select the ALSA AVB device and set number capture channels up to 8 and press record, and then try to playback through the ALSA AVB.

Sadly this does not seem to work. i can select the ABV device as output/input but when I hit play nothing happens, not even playhead moving through the waveform. the AVB device does not even show in the Pavucontrol available devices list. 

This is what I am getting when running avb_up.sh. 

`> $ sudo avb_up.sh enp14s0 96000                                                                                                                                                                                        
rmmod: ERROR: Module igb_avb is not currently loaded
driver: igb_avb
version: 5.3.2_AVB
firmware-version: 3.16, 0x800004ff, 1.304.0
expansion-rom-version: 
bus-info: 0000:0e:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no
Starting daemons on enp14s0
INFO     : GPTP [11:51:46:961] gPTP starting
ERROR    : GPTP [11:51:46:962] Using clock device: /dev/ptp0
STATUS   : GPTP [11:51:46:962] Starting PDelay
ERROR    : GPTP [11:51:47:058] Error (TX) timestamping PDelay request, error=-72
process_events()
STATUS   : GPTP [11:51:47:499] LINKUP
STATUS   : GPTP [11:51:47:500] Link Speed: 100000 kb/sec
STATUS   : GPTP [11:51:48:550] AsCapable: Enabled
STATUS   : GPTP [11:51:49:962] *** Announce Timeout Expired - Becoming Master
STATUS   : GPTP [11:51:49:962] New Grandmaster "6C:B3:11:FF:FE:52:67:23" (previous "00:00:00:00:00:00:00:00")
index=7
rc = 0, mrp_register_domain
rc = 0, mrp_join_vlan
Setting samplerate to 96000.
PARSE: C=6,P=3,V=0002
PARSE: I=0002

VNE 0002 R=000000000000 VN/MT

VNE 0002 R=000000000000 VN/MT
 unhandled from mrpd
SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN

Done setting samplerate.

Mac: 6c-b3-11-52-67-23
S+L:L=0001f2009a5f0000, D=2
rc = 0, done sending ready
rc = 0, advertising stream
PARSE: L=0001f2009a5f0000, D=2
PARSE: S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

EVENT on STREAM ID=0001f2009a5f0000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 0001f2009a5f0000
listener: 000000000000
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker:
mode JOIN
stream: 6cb3115267230000
dst_mac: 91e0f000fe00
vid: 2
SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker 1f2fffe009a5f, entity 6cb311fffe526723
VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

VJO 0002 R=0001f2009a5f QA/IN

VJO 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN

VJO 0002 R=0001f2009a5f QA/IN

VJO 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
`

**Comment by siborg** [comment]
I think I have followed setup instructions carefully: 

I have i210 card:

0e:00.0 Ethernet controller: Intel Corporation I210 Gigabit Network Connection 

kernel is built with ptp, igb and i2c_algobit as modules (I am using 5.12.0-rc6 atm, haven't tried regular 5.11 or earlier versions though)

kernel_module/igb/avb-config.h. have been modified with 
OWN_MAC : set to my i210 card mac
AVB_DEVICE_SOURCE_MAC: set to Motu Ultralite AVB MAC
AVB_DEVICE_TALKER_MAC_BASE: set to Motu Ultralite AVB

compiles and installs fine. 
When I run the avb_up.sh (with sudo, and nic and sample-rate parameters). I can see AVB synchronisation happening on Motu Ultralite AVB screen and the AVB device shows in WebConsole - 7 inputs are showing in Device section under AVB:0 device , no Output AVB devices are shown though - not sure if this is significant.  I can see 2 x 8 AVB streams in the routing section though.

**Comment by ericfont** [comment]
> the AVB device shows in WebConsole - 7 inputs are showing in Device section under AVB:0 device , no Output AVB devices are shown though - not sure if this is significant

Go ahead and in you motu webpage for routing, make sure for AVB ouput streams enter 1 too.  You should sees the 8-channel output stream and 8-channel input stream in that routing page even before starting AVB_up.sh

**Comment by siborg** [comment]
> > the AVB device shows in WebConsole - 7 inputs are showing in Device section under AVB:0 device , no Output AVB devices are shown though - not sure if this is significant
> 
> Go ahead and in you motu webpage for routing, make sure for AVB ouput streams enter 1 too. You should sees the 8-channel output stream and 8-channel input stream in that routing page even before starting AVB_up.sh

Sadly no luck. Yes, I can see the AVB streams and I made the routing connection and started avb shell script afterwards but still experiencing the same behaviou. I am going to change over to kernel 5.11 and try it instead, to rule out that it is the 5.12-rc6 which is causing the issue. Doesn't matter what I do I cannot play anything through that AVB:0 device, from any app.

**Comment by ericfont** [comment]
it is generally a bad idea to try experimental code on a release-candidate kernel, so can better narrow down where the error is from.

Also fyi you will have to recompile everytime you switch kernels, so it might be a good idea to stick on a lts kernel.

Though I have been successful on 5.10.4 (the kernel the developer's commits says its updated for), 5.10.27-lts, 5.11.4-rt, and 5.11.11.

**Comment by siborg** [comment]
No luck with with 5.11.11-arch1. Re-compiled everything - reinstalled. Same results. 

Can see AVB device:

>  aplay -l
> 
> card 7: AVB [AVB], device 0: AVB [AVB]
>   Subdevices: 1/1
>   Subdevice #0: subdevice #0


> aplay -L
> 
> usbstream:CARD=AVB
>     AVB
>     USB Stream Output

Screenshots of my AVB interface are attached. 

Lost for now as to what it could be. 

![screenshot](https://user-images.githubusercontent.com/2221778/113922902-c85d7000-97df-11eb-9a7a-d714f85b0a73.jpg)
![screenshot2](https://user-images.githubusercontent.com/2221778/113922904-c8f60680-97df-11eb-90e5-a8a4c14bc5a1.jpg)
![screenshot3](https://user-images.githubusercontent.com/2221778/113922905-c98e9d00-97df-11eb-975b-9fbce64f66c7.jpg)

I am getting the below entries in my syslog:

> Apr 07 19:03:32 gorzelnia kernel: audit: type=1111 audit(1617818612.559:211): pid=748 uid=0 auid=4294967295 ses=4294967295 msg='op=statistics arg="refresh-rate-ms" pid=2063 uid=1000 result=success exe="/usr/bin/Netwo>
> Apr 07 19:03:32 gorzelnia NetworkManager[748]: <info>  [1617818612.5635] device (enp14s0): state change: unmanaged -> unavailable (reason 'managed', sys-iface-state: 'external')
> Apr 07 19:03:32 gorzelnia systemd-udevd[42448]: Using default interface naming scheme 'v247'.
> Apr 07 19:03:33 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00001043
> Apr 07 19:03:33 gorzelnia systemd-networkd[41985]: enp14s0: Link UP
> Apr 07 19:03:33 gorzelnia kernel: probing avb dev
> Apr 07 19:03:33 gorzelnia kernel: enabling avb dev
> Apr 07 19:03:33 gorzelnia kernel: creating card avb dev
> Apr 07 19:03:33 gorzelnia kernel: creating pcm avb dev
> Apr 07 19:03:33 gorzelnia kernel: creating new pcm avb
> Apr 07 19:03:33 gorzelnia kernel: err 0
> Apr 07 19:03:33 gorzelnia kernel: prealloc pages of new pcm avb
> Apr 07 19:03:33 gorzelnia kernel: alloc tx new pcm avb
> Apr 07 19:03:33 gorzelnia kernel: alloc rx new pcm avb
> Apr 07 19:03:33 gorzelnia kernel: setting names pcm avb dev
> Apr 07 19:03:33 gorzelnia kernel: register card pcm avb dev
> Apr 07 19:03:33 gorzelnia kernel: register card pcm avb dev done
> Apr 07 19:03:33 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00001003
> Apr 07 19:03:33 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00001002
> Apr 07 19:03:33 gorzelnia systemd-networkd[41985]: enp14s0: Link DOWN
> Apr 07 19:03:33 gorzelnia systemd-udevd[42448]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
> Apr 07 19:03:33 gorzelnia kmix[1948]: UdevQt: unhandled device action "move"
> Apr 07 19:03:33 gorzelnia kdeconnectd[1919]: UdevQt: unhandled device action "move"
> Apr 07 19:03:33 gorzelnia org_kde_powerdevil[1877]: UdevQt: unhandled device action "move"
> Apr 07 19:03:33 gorzelnia kded5[1784]: UdevQt: unhandled device action "move"
> Apr 07 19:03:33 gorzelnia systemd-udevd[42397]: controlC7: /usr/lib/udev/rules.d/78-sound-card.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/0000:00:03.2/0000:0e:00.0/sound/card7/controlC7/../uevent}, ignoring>
> Apr 07 19:03:33 gorzelnia plasmashell[1874]: UdevQt: unhandled device action "move"
> Apr 07 19:03:33 gorzelnia systemd[1724]: Reached target Sound Card.
> Apr 07 19:03:34 gorzelnia kmix[1948]: org.kde.kmix: Cannot add mixer to GUI
> Apr 07 19:03:34 gorzelnia kernel: probing avb dev
> Apr 07 19:03:34 gorzelnia kernel: enabling avb dev
> Apr 07 19:03:34 gorzelnia kernel: creating card avb dev
> Apr 07 19:03:34 gorzelnia kernel: creating pcm avb dev
> Apr 07 19:03:34 gorzelnia kernel: creating new pcm avb
> Apr 07 19:03:34 gorzelnia kernel: err 0
> Apr 07 19:03:34 gorzelnia kernel: prealloc pages of new pcm avb
> Apr 07 19:03:34 gorzelnia kernel: alloc tx new pcm avb
> Apr 07 19:03:34 gorzelnia kernel: alloc rx new pcm avb
> Apr 07 19:03:34 gorzelnia kernel: setting names pcm avb dev
> Apr 07 19:03:34 gorzelnia kernel: register card pcm avb dev
> Apr 07 19:03:34 gorzelnia kernel: register card pcm avb dev done
> Apr 07 19:03:34 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00001003
> Apr 07 19:03:34 gorzelnia systemd-networkd[41985]: enp14s0: Link UP

> Apr 07 19:03:34 gorzelnia audit[748]: USYS_CONFIG pid=748 uid=0 auid=4294967295 ses=4294967295 msg='op=statistics arg="refresh-rate-ms" pid=2063 uid=1000 result=success exe="/usr/bin/NetworkManager" hostname=? addr=?>
> Apr 07 19:03:34 gorzelnia NetworkManager[748]: <info>  [1617818614.7887] audit: op="statistics" arg="refresh-rate-ms" pid=2063 uid=1000 result="success"
> Apr 07 19:03:34 gorzelnia audit: ANOM_PROMISCUOUS dev=enp14s0 prom=256 old_prom=0 auid=1000 uid=0 gid=0 ses=2
> Apr 07 19:03:34 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00001103
> Apr 07 19:03:34 gorzelnia audit[42461]: SYSCALL arch=c000003e syscall=16 success=yes exit=0 a0=4 a1=8914 a2=7ffd5ba48e20 a3=11 items=0 ppid=42358 pid=42461 auid=1000 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fs>
> Apr 07 19:03:34 gorzelnia audit: PROCTITLE proctitle=6966636F6E66696700656E70313473300075700070726F6D697363
> Apr 07 19:03:34 gorzelnia ksystemstats[2063]: UdevQt: unhandled device action "move"
> Apr 07 19:03:34 gorzelnia kernel: audit: type=1111 audit(1617818614.786:212): pid=748 uid=0 auid=4294967295 ses=4294967295 msg='op=statistics arg="refresh-rate-ms" pid=2063 uid=1000 result=success exe="/usr/bin/Netwo>
> Apr 07 19:03:34 gorzelnia kernel: device enp14s0 entered promiscuous mode
> Apr 07 19:03:34 gorzelnia kernel: audit: type=1700 audit(1617818614.786:213): dev=enp14s0 prom=256 old_prom=0 auid=1000 uid=0 gid=0 ses=2
> Apr 07 19:03:34 gorzelnia kernel: audit: type=1300 audit(1617818614.786:213): arch=c000003e syscall=16 success=yes exit=0 a0=4 a1=8914 a2=7ffd5ba48e20 a3=11 items=0 ppid=42358 pid=42461 auid=1000 uid=0 gid=0 euid=0 s>
> Apr 07 19:03:34 gorzelnia kernel: audit: type=1327 audit(1617818614.786:213): proctitle=6966636F6E66696700656E70313473300075700070726F6D697363
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to find a working profile.
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to find a working profile.
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to find a working profile.
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to find a working profile.
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia kernel: samplerate change not supported yet
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to find a working profile.
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Failed to load module "module-alsa-card" (argument: "device_id="7" name="pci-0000_0e_00.0" card_name="alsa_card.pci-0000_0e_00.0" namereg_fail=false tsched=yes fixed_latenc>
> Apr 07 19:03:34 gorzelnia pulseaudio[2222]: Tried to configure /devices/pci0000:00/0000:00:03.2/0000:0e:00.0/sound/card7 (alsa_card.pci-0000_0e_00.0) more often than 5 times in 10s
> Apr 07 19:03:35 gorzelnia kwin_x11[1788]: qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 59124, resource id: 35651888, major code: 15 (QueryTree), minor code: 0
> Apr 07 19:03:35 gorzelnia plasmashell[1874]: file:///home/skosecki/.local/share/plasma/wallpapers/com.github.zren.inactiveblur/contents/ui/WindowModel.qml:52: Error: Invalid write to global property "currentWindowMax>
> Apr 07 19:03:35 gorzelnia kmix[1948]: org.kde.kmix: no mix devices and not dynamic
> Apr 07 19:03:35 gorzelnia kmix[1948]: org.kde.kmix: Cannot add mixer to GUI
> Apr 07 19:03:35 gorzelnia kded5[1784]: org.kde.kmix: no mix devices and not dynamic
> Apr 07 19:03:35 gorzelnia kwin_x11[1788]: qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 59344, resource id: 35651894, major code: 15 (QueryTree), minor code: 0
> Apr 07 19:03:35 gorzelnia plasmashell[1874]: file:///home/skosecki/.local/share/plasma/wallpapers/com.github.zren.inactiveblur/contents/ui/WindowModel.qml:52: Error: Invalid write to global property "currentWindowMax>
> Apr 07 19:03:36 gorzelnia kernel: igb_avb 0000:0e:00.0 enp14s0: igb: enp14s0 NIC Link is Up 100 Mbps Full Duplex, Flow Control: RX/TX
> Apr 07 19:03:36 gorzelnia kernel: IPv6: ADDRCONF(NETDEV_CHANGE): enp14s0: link becomes ready
> Apr 07 19:03:36 gorzelnia systemd-networkd[41985]: enp14s0: Gained carrier
> Apr 07 19:03:36 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00011143
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4137] device (enp14s0): carrier: link connected
> Apr 07 19:03:36 gorzelnia avahi-daemon[744]: Joining mDNS multicast group on interface enp14s0.IPv4 with address 192.168.168.1.
> Apr 07 19:03:36 gorzelnia avahi-daemon[744]: New relevant interface enp14s0.IPv4 for mDNS.
> Apr 07 19:03:36 gorzelnia avahi-daemon[744]: Registering new address record for 192.168.168.1 on enp14s0.IPv4.
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4150] device (enp14s0): state change: unavailable -> disconnected (reason 'carrier-changed', sys-iface-state: 'managed')
> Apr 07 19:03:36 gorzelnia vmnetBridge[823]: RTM_NEWLINK: name:enp14s0 index:6 flags:0x00011143
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4180] device (enp14s0): Activation: starting connection 'enp14s0' (59dfa956-b445-4b37-928d-90cc7cbdb5bf)
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4187] device (enp14s0): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4189] device (enp14s0): state change: prepare -> config (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4192] device (enp14s0): state change: config -> ip-config (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4194] device (enp14s0): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4274] device (enp14s0): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4275] device (enp14s0): state change: secondaries -> activated (reason 'none', sys-iface-state: 'external')
> Apr 07 19:03:36 gorzelnia NetworkManager[748]: <info>  [1617818616.4283] device (enp14s0): Activation: successful, device activated.
> Apr 07 19:03:42 gorzelnia systemd[1]: NetworkManager-dispatcher.service: Deactivated successfully.

So it looks like pulseaudio is trying to claim it but it fails in the end. Looks like for whatever reason the PCM device is never reaching working state.

Quick question, when modyfing the kernel-module/igb/abv-config.h did you leave the trailing **ul** at the end of the mac address in the config, or just used the raw mac address (: removed obviously).?

**Comment by ericfont** [comment]
> Quick question, when modyfing the kernel-module/igb/abv-config.h did you leave the trailing **ul** at the end of the mac address in the config, or just used the raw mac address (: removed obviously).?

I left the trailing **ul** on after the end of the mac address.  It tells the compiler that the number should be treated as "unsigned long" (a 32-bit number without a positive or negative sign)

**Comment by ericfont** [comment]
> So it looks like pulseaudio is trying to claim it but it fails in the end. Looks like for whatever reason the PCM device is never reaching working state.

I thought I told you to use pavucontrol and find the AVB network device, and set it to "OFF" or otherwise disable pulseaudio so that pulseaudio doesn't take control of the device:

![pavu-settings](https://user-images.githubusercontent.com/6502474/113938624-dc49a780-97c8-11eb-9d9a-46638056553a.png)

**Comment by siborg** [comment]
> > So it looks like pulseaudio is trying to claim it but it fails in the end. Looks like for whatever reason the PCM device is never reaching working state.
> 
> I thought I told you to use pavucontrol and find the AVB network device, and set it to "OFF" or otherwise disable pulseaudio so that pulseaudio doesn't take control of the device:
> 
> ![pavu-settings](https://user-images.githubusercontent.com/6502474/113938624-dc49a780-97c8-11eb-9d9a-46638056553a.png)

Indeed you have, however AVB device does not show in pavucontrol for me. I only see other devices. 

![screenshot](https://user-images.githubusercontent.com/2221778/113939106-7292c300-97f3-11eb-9350-f168b4a4e080.jpg)

**Comment by ericfont** [comment]
One piece of advice, is keep your setup simple when trying to get this to work for the first time.  I notice your Motu web interface shows have two AVB devices, one called "AVB" and the Ultralite, and you have 2 input streams and 2 output streams:

![image](https://user-images.githubusercontent.com/6502474/113951416-82a1a700-97e1-11eb-8c2d-7f874080955d.png)

I actually don't know what that "AVB" device is.  When I use my UltraLite alone on my network, I only see the ultralite listed in my motu web interface, and when I first set it up, I used only 1 input stream and 1 output stream.  So try disconnecting that other AVB device you have other than your UltraLite.  And then use wireshark, and make sure you pick up the mac address of the ultralite correctly and its MAAP MAC address for the stream and entered correclty into avb_config.h before recompiling again (run `sudo wireshark -i device -k` and input the display filter for `maap`).

**Comment by siborg** [comment]
The thing is I am not doing anything special to configure it. It shows after I run avb_up.sh. I will do the wireshark dump. Thank you for the good pointers!

**Comment by Drumfix** [comment]
The AVB device is your computer

**Comment by ericfont** [comment]
> The AVB device is your computer

I wonder why my computer doesn't display as "AVB" there in the left-column under devices:

![image](https://user-images.githubusercontent.com/6502474/114211381-8ee74a80-992e-11eb-8e27-5fb75b82f364.png)

And I wonder why under "INPUT CONNECTIONS", my computer AVB is connected with the misleading name "Offline Stream".

**Comment by siborg** [comment]
We're using different devices I suppose with probably slightly different
firmwares, perhaps this can account for the differences.
Very Best Regards,

Szymon Kosecki

**Comment by ericfont** [comment]
ah, good point.  For the record, my Motu 624 is on Firmware version 1.3.5+226, GUI version 1.1.1+TEST

![image](https://user-images.githubusercontent.com/6502474/114213557-1a61db00-9931-11eb-8106-2244b8bb80c5.png)

**Comment by ericfont** [comment]
I wonder if there is a way to get the computer to display as "avb4linux" rather than "AVB".  Unless motu is using their own non-open standard for how devices are named.

**Comment by siborg** [comment]
I am running:

[image: image.png]

Very Best Regards,

Szymon Kosecki

**Comment by Drumfix** [comment]
The name you see is the name given in avb-user.c - fill_jdksavdecc_descriptor_entity - entity_name

**Comment by siborg** [comment]
Hello Drumfix,

Were you able to get it working wit Ultralite AVB by any chance? I am
struggling with the PCM device not really working. It is created, AVB
detects the new AVB device but I cannot either play any audio through or
start Jack. Pulseaudio is disabled.


Very Best Regards,

Szymon Kosecki

**Comment by ericfont** [comment]
(I know you asked the drumfix...but I can confirm that avb4linux works both with my Motu UltraLite AVB in addition to my Motu 624 AVB)

**Comment by Drumfix** [comment]
Have you updated avb4linux to the latest version?
And make sure for the moment that the device is not connected to a USB port.

If you see (in your case)

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN

VJO 0002 R=0001f2009a5f QA/IN

VJO 0002 R=0001f2009a5f QA/IN
unhandled from mrpd
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN


then the input and output streams are connected and streaming.

**Comment by siborg** [comment]
> Have you updated avb4linux to the latest version?
> And make sure for the moment that the device is not connected to a USB port.
> 
> If you see (in your case)
> 
> talker:
> mode JOIN
> stream: 0001f2009a5f0000
> dst_mac: 91e0f0002a2c
> vid: 2
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> EVENT on STREAM ID=6cb3115267230000 with state ready
> got a new/join indication
> listener:
> mode JOIN
> stream: 6cb3115267230000
> listener: 0001f2009a5f
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> unhandled from mrpd
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> then the input and output streams are connected and streaming.

I am seeing the following output:

```
> $ sudo avb_up.sh enp14s0 96000                                                                                                                                                                                        
rmmod: ERROR: Module igb_avb is not currently loaded
driver: igb_avb
version: 5.3.2_AVB
firmware-version: 3.16, 0x800004ff, 1.304.0
expansion-rom-version: 
bus-info: 0000:0e:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no
Starting daemons on enp14s0
INFO     : GPTP [15:32:27:313] gPTP starting
ERROR    : GPTP [15:32:27:314] Using clock device: /dev/ptp0
STATUS   : GPTP [15:32:27:314] Starting PDelay
ERROR    : GPTP [15:32:27:409] Error (TX) timestamping PDelay request, error=-72
process_events()
STATUS   : GPTP [15:32:27:942] LINKUP
STATUS   : GPTP [15:32:27:943] Link Speed: 100000 kb/sec
STATUS   : GPTP [15:32:29:008] AsCapable: Enabled
STATUS   : GPTP [15:32:30:314] *** Announce Timeout Expired - Becoming Master
STATUS   : GPTP [15:32:30:314] New Grandmaster "6C:B3:11:FF:FE:52:67:23" (previous "00:00:00:00:00:00:00:00")
index=7
rc = 0, mrp_register_domain
rc = 0, mrp_join_vlan
Setting samplerate to 96000.
PARSE: C=6,P=3,V=0002
PARSE: I=0002

VNE 0002 R=000000000000 VN/MT

VNE 0002 R=000000000000 VN/MT
 unhandled from mrpd
SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN

Done setting samplerate.

Mac: 6c-b3-11-52-67-23
S+L:L=0001f2009a5f0000, D=2
rc = 0, done sending ready
rc = 0, advertising stream
PARSE: L=0001f2009a5f0000, D=2
PARSE: S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

EVENT on STREAM ID=0001f2009a5f0000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 0001f2009a5f0000
listener: 000000000000
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker:
mode JOIN
stream: 6cb3115267230000
dst_mac: 91e0f000fe00
vid: 2
SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker 1f2fffe009a5f, entity 6cb311fffe526723
VNE 0002 R=0001f2009a5f AA/IN

VNE 0002 R=0001f2009a5f AA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

VJO 0002 R=0001f2009a5f QA/IN

VJO 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

```

**Comment by siborg** [comment]
> Have you updated avb4linux to the latest version?
> And make sure for the moment that the device is not connected to a USB port.
> 
> If you see (in your case)
> 
> talker:
> mode JOIN
> stream: 0001f2009a5f0000
> dst_mac: 91e0f0002a2c
> vid: 2
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> EVENT on STREAM ID=6cb3115267230000 with state ready
> got a new/join indication
> listener:
> mode JOIN
> stream: 6cb3115267230000
> listener: 0001f2009a5f
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> unhandled from mrpd
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> then the input and output streams are connected and streaming.

I am seeing the following output:

> $ sudo avb_up.sh enp14s0 96000                                                                                                                                                                                        
rmmod: ERROR: Module igb_avb is not currently loaded
driver: igb_avb
version: 5.3.2_AVB
firmware-version: 3.16, 0x800004ff, 1.304.0
expansion-rom-version: 
bus-info: 0000:0e:00.0
supports-statistics: yes
supports-test: yes
supports-eeprom-access: yes
supports-register-dump: yes
supports-priv-flags: no
Starting daemons on enp14s0
INFO     : GPTP [15:32:27:313] gPTP starting
ERROR    : GPTP [15:32:27:314] Using clock device: /dev/ptp0
STATUS   : GPTP [15:32:27:314] Starting PDelay
ERROR    : GPTP [15:32:27:409] Error (TX) timestamping PDelay request, error=-72
process_events()
STATUS   : GPTP [15:32:27:942] LINKUP
STATUS   : GPTP [15:32:27:943] Link Speed: 100000 kb/sec
STATUS   : GPTP [15:32:29:008] AsCapable: Enabled
STATUS   : GPTP [15:32:30:314] *** Announce Timeout Expired - Becoming Master
STATUS   : GPTP [15:32:30:314] New Grandmaster "6C:B3:11:FF:FE:52:67:23" (previous "00:00:00:00:00:00:00:00")
index=7
rc = 0, mrp_register_domain
rc = 0, mrp_join_vlan
Setting samplerate to 96000.
PARSE: C=6,P=3,V=0002
PARSE: I=0002

VNE 0002 R=000000000000 VN/MT

VNE 0002 R=000000000000 VN/MT
 unhandled from mrpd
SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN

Done setting samplerate.

Mac: 6c-b3-11-52-67-23
S+L:L=0001f2009a5f0000, D=2
rc = 0, done sending ready
rc = 0, advertising stream
PARSE: L=0001f2009a5f0000, D=2
PARSE: S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

EVENT on STREAM ID=0001f2009a5f0000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 0001f2009a5f0000
listener: 000000000000
SNE L:D=2,S=0001f2009a5f0000 R=000000000000 VN/MT

SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker:
mode JOIN
stream: 6cb3115267230000
dst_mac: 91e0f000fe00
vid: 2
SNE T:S=6cb3115267230000,A=91e0f000fe00,V=0002,Z=416,I=1,P=112,L=95 R=000000000000 VN/MT

talker 1f2fffe009a5f, entity 6cb311fffe526723
VNE 0002 R=0001f2009a5f AA/IN

VNE 0002 R=0001f2009a5f AA/IN
 unhandled from mrpd
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SNE L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

VNE 0002 R=0001f2009a5f QA/IN

VNE 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SNE T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

VJO 0002 R=0001f2009a5f QA/IN

VJO 0002 R=0001f2009a5f QA/IN
 unhandled from mrpd
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

talker:
mode JOIN
stream: 0001f2009a5f0000
dst_mac: 91e0f0002a2c
vid: 2
SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN

SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN

EVENT on STREAM ID=6cb3115267230000 with state ready
got a new/join indication
listener:
mode JOIN
stream: 6cb3115267230000
listener: 0001f2009a5f
SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN



> Have you updated avb4linux to the latest version?
> And make sure for the moment that the device is not connected to a USB port.
> 
> If you see (in your case)
> 
> talker:
> mode JOIN
> stream: 0001f2009a5f0000
> dst_mac: 91e0f0002a2c
> vid: 2
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> EVENT on STREAM ID=6cb3115267230000 with state ready
> got a new/join indication
> listener:
> mode JOIN
> stream: 6cb3115267230000
> listener: 0001f2009a5f
> SJO L:D=2,S=6cb3115267230000 R=0001f2009a5f VO/IN
> 
> SJO D:C=6,P=3,V=0002,N=3 R=0001f2009a5f AA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> 
> VJO 0002 R=0001f2009a5f QA/IN
> unhandled from mrpd
> SJO T:S=0001f2009a5f0000,A=91e0f0002a2c,V=0002,Z=416,I=1,P=112,L=95 R=0001f2009a5f VO/IN
> 
> then the input and output streams are connected and streaming.

Quick update, there is some improvement after pulling the latest git repo changes and recompiling. I can at least see the PCM device in PulseAudio now (previously it could never claim the device and was not showing it). 

I've rebooted, disabled pulse audio, started the avb, run the script, but still I am unable to get any sound from the PC to the Ultralite. 

Audacity configured to use AVB:0 device is unable to start playback (the playhead won't even move) and Jack also does not start (8 in and 8 out channels configured). 

Ultralite is not connected via USB at all. Only cat 5e cable.

**Comment by Drumfix** [comment]
Somehow i suspect your values in avb-config.h are  wrong

Looking at your data, you should have

#define OWN_MAC 0x6cb311526723ul
#define AVB_DEVICE_SOURCE_MAC 0x0001f2009a5ful
#define AVB_DEVICE_TALKER_MAC_BASE 0x91e0f0002a2cul

**Comment by siborg** [comment]
Thanks Drumfix. I had AVB_DEVICE_TALKER_MAC_BASE set up the same as the
Ultralite MAC.
Does it relate to MAAP_LOCALLY_ADMINISTRATORED_BASE

Very Best Regards,

Szymon Kosecki

**Comment by siborg** [comment]
> Somehow i suspect your values in avb-config.h are wrong
> 
> Looking at your data, you should have
> 
> #define OWN_MAC 0x6cb311526723ul
> #define AVB_DEVICE_SOURCE_MAC 0x0001f2009a5ful
> #define AVB_DEVICE_TALKER_MAC_BASE 0x91e0f0002a2cul

Thanks a lot! I have audio now and jack starts. It's cracking very often (but no xruns). Any particular jack settings that work well for you that we should be aware of?

**Comment by Drumfix** [comment]
If you have crackling you could try to increase the  PRESENTATION_TIME_OFFSET in also.c

You should try in increments of 100000, although the default of 0 should be fine with the Ultralite AVB.
If this doesn't help you need to look at your realtime  priorites. The interrupt with -TxRx-0 in the name must have the highest priority (if you are using threaded interrupts)

**Comment by siborg** [comment]
Thanks Drumfix.  PRESENTATION_TIME_OFFSET doesn't do the trick I am afraid.
yes I am running my kernel with threadirqs as standard.
Any particular name I need to add to rtirq.conf to make that priority
setting happen? Apologies, if I am asking questions all the time.


Very Best Regards,

Szymon Kosecki

**Comment by Drumfix** [comment]
i guess it is enp14s0-TxRx-0.

just do a "cat /proc/interrupts"

You may also need to set priorites in pulseaudio (that is somewhere in /etc/pulseaudio) and jack.
Both must have a lower priority than the interrupt.

**Comment by siborg** [comment]
> i guess it is enp14s0-TxRx-0.
> 
> just do a "cat /proc/interrupts"
> 
> You may also need to set priorites in pulseaudio (that is somewhere in /etc/pulseaudio) and jack.
> Both must have a lower priority than the interrupt.

Thanks a lot, will experiment with various combinations! One last question, realistically with  PRESENTATION_TIME_OFFSET , how big this value can be? I've tried up to 600000 in 100000 increments.

**Comment by Drumfix** [comment]
Dunno. The resulting presentation time /.e. presentation time received in the packet from the device + PRESENTATION_TIME_OFFSET) must not be greater than 2 milliseconds. I have never looked at what the presentation time in the packet from the device actually is. I wouldn't go over 300000.

Can we close the issue, because the crackling shouldn't be related to the driver working correctly?

**Comment by siborg** [comment]
Thank you, please close the issue. it is probably peculiarity of my
hardware causing this. Once more, you're the man!


Very Best Regards,

Szymon Kosecki
